vault/resource_ssh_secret_backend_ca.go

```diff
@@ -0,0 +1,140 @@
+package vault
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"log"
+	"strings"
+)
+
+func sshSecretBackendCAResource() *schema.Resource {
+	return &schema.Resource{
+		Create: sshSecretBackendCACreate,
+		Read:   sshSecretBackendCARead,
+		Delete: sshSecretBackendCADelete,
+		Exists: sshSecretBackendCAExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "ssh",
+				ForceNew:    true,
+				Description: "The path of the SSH Secret Backend where the CA should be configured",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"generate_signing_key": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				ForceNew:    true,
+				Description: "Whether Vault should generate the signing key pair internally.",
+			},
+			"private_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Computed:    true,
+				Description: "Private key part the SSH CA key pair; required if generate_signing_key is false.",
+			},
+			"public_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Public key part the SSH CA key pair; required if generate_signing_key is false.",
+			},
+		},
+	}
+}
+
+func sshSecretBackendCACreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	generateSigningKey := d.Get("generate_signing_key").(bool)
+
+	data := map[string]interface{}{
+		"generate_signing_key": generateSigningKey,
+	}
+	if !generateSigningKey {
+		privateKey := d.Get("private_key").(string)
+		publicKey := d.Get("public_key").(string)
+		if privateKey == "" || publicKey == "" {
+			return fmt.Errorf("When generate_signing_key is false, both public_key and private_keys should be specified")
```

Comment:
Is this error needed in light of how the code linked above actually does allow this case? 

It's useful, though, because we'd only want to populate the public and private key if they were actually provided, in this particular case.

Reply:
See previous comment.